llvm/lib/Analysis/StackSafetyAnalysis.cpp

```diff
@@ -268,13 +276,28 @@ class StackSafetyLocalAnalysis {
   FunctionInfo<GlobalValue> run();
 };
 
+const SCEV *StackSafetyLocalAnalysis::getSCEVAsPointer(Value *Val) {
+  Type *ValTy = Val->getType();
+
+  auto *PtrTy = PointerType::getUnqual(SE.getContext());
+  if (ValTy->isPointerTy()) {
```

Comment:
Is this ever false? I don't think this code is working on anything but pointers.

Reply:
yeah it only works on pointers, I added an assert and it doesn't trip